client/components/common/customHooks/useOutsideClick.js

```diff
@@ -0,0 +1,18 @@
+import { useEffect } from 'react';
+
+const useOutsideClick = (ref, callback) => {
```

Comment:
Nice, this is great. Going to add it to the RequestTypeSelector after this is merged in.

Reply:
Maybe we can add it to the selector box as well.